### PR TITLE
[ISSUE-#4258]  Fix wrong path when -Dspring.config.location not set

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change

Fix-#4258

### CODE REVIEW

File: sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java [hunk 1]
```diff
@@ -361,19 +361,35 @@ public static String getMemberList() {
...
         if (StringUtils.isNotBlank(path) && path.contains(FILE_PREFIX)) {
             String[] paths = path.split(",");
             path = paths[paths.length - 1].substring(FILE_PREFIX.length());
         }
+        if (!path.endsWith("/")) {
+            path += "/";
+        }
         try {
-            inputStream = new FileInputStream(new File(path + "application.properties"));
+            InputStream inputStream = new FileInputStream(new File(path + "application.properties"));
```
Comment (chuntaojun) [inline sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:385]: new File(path + "application.properties") => Paths.get().toFile()

Comment (horizonzy) [reply sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java]: ok

File: sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java [hunk 2]
```diff
@@ -363,15 +363,15 @@ public static String getMemberList() {
...
+            return getDefaultResource();
         }
         return customResource;
     }
     
     private static Resource getCustomFileResource() {
         String path = getProperty("spring.config.location");
         if (path == null) {
-            return null;
+            path = EnvUtil.getNacosHome() + "/conf";
```
Comment (chuntaojun) [inline sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:374]: use Paths.get(...)

**Review by chuntaojun — APPROVED** [review]

**Review by chuntaojun — DISMISSED** [review]

File: sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java [hunk 3]
```diff
@@ -361,19 +362,38 @@ public static String getMemberList() {
...
+        if (customResource == null) {
+            return getDefaultResource();
         }
+        return customResource;
+    }
+    
+    private static Resource getCustomFileResource() {
+        String customPath = getProperty("spring.config.location");
+        if (customPath == null) {
+            return getRelativePathResource(Paths.get(EnvUtil.getNacosHome(), "conf"));
```
Comment (Maijh97) [inline sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:375]: 这个`if (customPath == null)`可以去掉，因为方法结束会`return getRelativePathResource(Paths.get(EnvUtil.getNacosHome(), "conf"));`

**Review by Maijh97 — APPROVED** [review]

**Review by Maijh97 — CHANGES_REQUESTED** [review]
Solve CI error.

**Comment by horizonzy** [comment]
> Solve CI error.

I think it's CI mistake

File: sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java [hunk 4]
```diff
@@ -370,21 +369,18 @@ public static Resource getApplicationConfFileResource() {
...
     private static Resource getCustomFileResource() {
-        String customPath = getProperty("spring.config.location");
-        
-        if (StringUtils.isNotBlank(customPath) && customPath.contains(FILE_PREFIX)) {
-            String[] paths = customPath.split(",");
-            customPath = paths[paths.length - 1].substring(FILE_PREFIX.length());
-            Path parentPath = Paths.get(customPath);
-            return getRelativePathResource(parentPath);
+        String path = getProperty("spring.config.location");
+        if (StringUtils.isNotBlank(path) && path.contains(FILE_PREFIX)) {
```
[the remark this answers is not in the document]
Comment (horizonzy) [reply sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:373]: 现在有个containsIgnoreCase. 可以用这个替换

[the remark this answers is not in the document]
Comment (chuntaojun) [reply sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:373]: 路径不能IgnoreCase

[the remark this answers is not in the document]
Comment (horizonzy) [reply sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:373]: 这个只是判断`file:`, 这个大小写忽略ok的

**Review by chuntaojun — DISMISSED** [review]

**Review by Maijh97 — DISMISSED** [review]

File: sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java [hunk 5]
```diff
@@ -361,19 +361,34 @@ public static String getMemberList() {
     private static final String FILE_PREFIX = "file:";
     
     public static Resource getApplicationConfFileResource() {
+        Resource customResource = getCustomFileResource();
+        if (customResource == null) {
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/env/EnvUtil.java:365]: Simply to `null == customResource ? return getDefaultResource() : customResource`

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by Maijh97 — APPROVED** [review]

**Review by chuntaojun — APPROVED** [review]

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**